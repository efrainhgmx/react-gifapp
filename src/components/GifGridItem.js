import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, url, title } ) => {

    console.log( id, url, title )
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propsTypes = {
    id: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
