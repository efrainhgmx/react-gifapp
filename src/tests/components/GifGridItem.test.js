import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
//import toJson from 'enzyme-to-json';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test on GifGridItem component', () => {
    const id = 'idoksh455bns';
    const url = 'https://google.com/google-image.jpg';
    const title = 'Google';

    test('The component should be show correctly', () => {
        const wrapper = shallow(
            <GifGridItem id={id} url={url} title={title} />
        );

        expect(wrapper).toMatchSnapshot();
    });
    
})
