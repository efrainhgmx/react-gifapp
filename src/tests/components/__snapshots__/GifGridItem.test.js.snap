// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGridItem component The component should be show correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Google"
    src="https://google.com/google-image.jpg"
  />
  <p>
    Google
  </p>
</div>
`;
